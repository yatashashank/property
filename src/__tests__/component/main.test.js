import React from 'react';
import toJson from 'enzyme-to-json';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import { shallow,render,mount } from 'enzyme';

import Main from '../../Components/Property/main'

import db from '../../dp.json'

describe('<Main />', () => {

  it('snapshot testing for Main component', () => {
    const wrapper = shallow(<Main />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('class name check', () => {
    const wrapper = shallow(<Main />);
    const tree =wrapper.find('div.card')
    expect(tree.length).toBe(1)
  });

  it('state values ', () => {
    const wrapper = shallow(<Main />);
    const tree =wrapper.state().property;
    expect(tree.length).toBe(db.length)
  });

}) 