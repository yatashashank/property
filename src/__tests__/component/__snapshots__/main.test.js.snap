// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> snapshot testing for Main component 1`] = `
<div
  className="card"
>
  <WithStyles(withRouter(Property))
    id={1}
    key="1"
    prop={
      Object {
        "id": 1,
        "location": "2563 8th ave",
        "occupants": Array [
          Object {
            "age": 57,
            "name": "president",
          },
          Object {
            "age": 34,
            "name": "secretary",
          },
          Object {
            "age": 25,
            "name": "butler",
          },
        ],
        "pricepermonth": 2530,
        "property": "WhiteHouse",
      }
    }
  />
  <WithStyles(withRouter(Property))
    id={2}
    key="2"
    prop={
      Object {
        "id": 2,
        "location": "32 north street",
        "occupants": Array [
          Object {
            "age": 26,
            "name": "Bob",
          },
        ],
        "pricepermonth": 684,
        "property": "Bobs crab shack",
      }
    }
  />
  <WithStyles(withRouter(Property))
    id={3}
    key="3"
    prop={
      Object {
        "id": 3,
        "location": "7983 west big Rd",
        "occupants": Array [
          Object {
            "age": 34,
            "name": "Gerry",
          },
        ],
        "pricepermonth": 983,
        "property": "Gerrys Fishing Store",
      }
    }
  />
  <WithStyles(withRouter(Property))
    id={4}
    key="4"
    prop={
      Object {
        "id": 4,
        "location": "0382 witherton Ave",
        "occupants": Array [
          Object {
            "age": 78,
            "name": "anne",
          },
        ],
        "pricepermonth": 1020,
        "property": "Annes Rental Property",
      }
    }
  />
  <WithStyles(withRouter(Property))
    id={5}
    key="5"
    prop={
      Object {
        "id": 5,
        "location": "0382 witherton Ave",
        "occupants": Array [
          Object {
            "age": 78,
            "name": "jhon",
          },
        ],
        "pricepermonth": 230,
        "property": "jhon Property",
      }
    }
  />
  <WithStyles(withRouter(Property))
    id={6}
    key="6"
    prop={
      Object {
        "id": 6,
        "location": "0382 witherton Ave",
        "occupants": Array [
          Object {
            "age": 78,
            "name": "jhon",
          },
        ],
        "pricepermonth": 230,
        "property": "jhon Property",
      }
    }
  />
</div>
`;
